extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts

```diff
@@ -0,0 +1,374 @@
+		projectLocationLabel.updateCssStyles({ 'font-weight': 'bold' });
+
+		const projectLocationRow = view.modelBuilder.flexContainer().withItems([projectLocationLabel, this.projectLocationTextBox], { flex: '0 0 auto', CSSStyles: { 'margin-right': '10px', 'margin-bottom': '-10px' } }).withLayout({ flexFlow: 'row', alignItems: 'center' }).component();
+		projectLocationRow.insertItem(browseFolderButton, 2, { CSSStyles: { 'margin-right': '0px', 'margin-bottom': '-10px' } });
+
+		return projectLocationRow;
+	}
+
+	private createBrowseFolderButton(view: azdata.ModelView): azdata.ButtonComponent {
+		const browseFolderButton = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			ariaLabel: constants.browseButtonText,
+			iconPath: IconPathHelper.folder_blue,
+			height: '18px',
+			width: '18px'
+		}).component();
+
+		browseFolderButton.onDidClick(async () => {
+			let folderUris = await vscode.window.showOpenDialog({
+				canSelectFiles: false,
+				canSelectFolders: true,
+				canSelectMany: false,
+				openLabel: constants.selectString,
+				defaultUri: newProjectTool.defaultProjectSaveLocation()
+			});
+			if (!folderUris || folderUris.length === 0) {
+				return;
+			}
+
+			this.projectLocationTextBox!.value = folderUris[0].fsPath;
+			this.projectLocationTextBox!.updateProperty('title', folderUris[0].fsPath);
+		});
+
+		return browseFolderButton;
+	}
+
+	private createFolderStructureRow(view: azdata.ModelView): azdata.FlexContainer {
+		this.folderStructureDropDown = view.modelBuilder.dropDown().withProperties({
+			values: [constants.file, constants.flat, constants.objectType, constants.schema, constants.schemaObjectType],
+			value: constants.schemaObjectType,
+			ariaLabel: constants.folderStructureLabel,
+			required: true,
+			width: cssStyles.textboxWidth
+		}).component();
+
+		this.folderStructureDropDown.onValueChanged(() => {
+			this.tryEnableCreateButton();
+		});
+
+		const folderStructureLabel = view.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: constants.folderStructureLabel,
+			requiredIndicator: true,
+			width: cssStyles.labelWidth
+		}).component();
+		folderStructureLabel.updateCssStyles({ 'font-weight': 'bold' });
+
+		const folderStructureRow = view.modelBuilder.flexContainer().withItems([folderStructureLabel, <azdata.DropDownComponent>this.folderStructureDropDown], { flex: '0 0 auto', CSSStyles: { 'margin-right': '10px', 'margin-top': '-20px' } }).withLayout({ flexFlow: 'row', alignItems: 'center' }).component();
+
+		return folderStructureRow;
+	}
+
+	// only enable Create button if all fields are filled
+	public tryEnableCreateButton(): void {
+		if (this.sourceConnectionTextBox!.value && this.sourceDatabaseDropDown!.value
+			&& this.projectNameTextBox!.value && this.projectLocationTextBox!.value) {
+			this.dialog.okButton.enabled = true;
+		} else {
+			this.dialog.okButton.enabled = false;
+		}
+	}
+
+	public async createButtonClick(): Promise<void> {
```

Comment:
the name is weird, how about handleOkButtonClick

Reply:
Updated it to "handleCreateButtonClick"